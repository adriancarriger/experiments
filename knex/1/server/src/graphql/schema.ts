import { gql } from 'apollo-server';

import book from './Book/schema';
import customer from './Customer/schema';

const base = gql`
  type Query
`;

export default [base, book, customer];
