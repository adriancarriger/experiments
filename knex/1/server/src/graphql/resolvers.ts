import { merge } from 'lodash';

import bookResolvers from './Book/resolvers';
import customerResolvers from './Customer/resolvers';

export default merge(bookResolvers, customerResolvers);
