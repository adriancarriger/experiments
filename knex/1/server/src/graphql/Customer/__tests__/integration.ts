import { gql } from 'apollo-server';

import customerResolvers from '../resolvers';
import customerSchema from '../schema';

import { createClient, disconnectDb, Client } from '../../__tests__/graphql.helpers';

const SEARCH_CUSTOMERS = gql`
  query {
    getCustomers(first_name: "Andy") {
      first_name
      last_name
    }
  }
`;

const GET_CUSTOMER = gql`
  query {
    getCustomer(id: "1") {
      first_name
      last_name
    }
  }
`;

const ADD_CUSTOMER = gql`
  mutation {
    addCustomer(customer: { first_name: "Ron", last_name: "Swanson" }) {
      id
      first_name
      last_name
    }
  }
`;

const UDATE_CUSTOMER = gql`
  mutation($id: ID!, $first_name: String!, $last_name: String!) {
    updateCustomer(id: $id, customer: { first_name: $first_name, last_name: $last_name }) {
      id
      first_name
    }
  }
`;

describe('Queries', () => {
  let client: Client;

  beforeAll(() => {
    client = createClient([customerSchema], customerResolvers);
  });

  afterAll(disconnectDb);

  it('fetches customers', async () => {
    const { data } = await client.query({ query: SEARCH_CUSTOMERS });
    expect(data).toMatchSnapshot();
  });

  it('fetches a specific customer', async () => {
    const { data } = await client.query({ query: GET_CUSTOMER });
    expect(data).toMatchSnapshot();
  });

  it('adds a customer', async () => {
    const {
      data: { addCustomer }
    } = await client.mutate({ query: ADD_CUSTOMER });
    expect(addCustomer.first_name).toBe('Ron');

    const {
      data: { updateCustomer }
    } = await client.mutate({
      query: UDATE_CUSTOMER,
      variables: { ...addCustomer, first_name: 'Diane' }
    });

    expect(updateCustomer.first_name).toBe('Diane');
  });
});
