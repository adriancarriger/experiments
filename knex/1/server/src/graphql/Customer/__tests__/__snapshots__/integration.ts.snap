// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches a specific customer 1`] = `
Object {
  "getCustomer": Object {
    "first_name": "Chris",
    "last_name": "Traeger",
  },
}
`;

exports[`Queries fetches customers 1`] = `
Object {
  "getCustomers": Array [
    Object {
      "first_name": "Andy",
      "last_name": "Dwyer",
    },
  ],
}
`;
